src/facebook_to_engagement_db/configuration.py

```diff
@@ -0,0 +1,49 @@
+from dateutil.parser import isoparse
+
+
+class FacebookSource:
+    def __init__(self, page_id, token_file_url, datasets):
+        self.page_id = page_id
+        self.token_file_url = token_file_url
+        self.datasets = datasets
+
+    def to_dict(self):
+        return {
+            "page_id": self.page_id,
+            "token_file_url": self.token_file_url,
+            "datasets": [dataset for dataset in self.datasets]
+        }
+
+
+class FacebookDataset:
+    def __init__(self, engagement_db_dataset, post_ids=None, search=None):
+        self.engagement_db_dataset = engagement_db_dataset
+        self.post_ids = post_ids
+        self.search = search
+
+        if search is not None:
+            self.search = FacebookSearch.to_dict(search)
```

Comment:
I think this should be in the `to_dict` function

Reply:
This is cleaner to avoid abstracting the to_dict function.